targets/tinyfpga_bx/internal.py

```diff
@@ -0,0 +1,177 @@
+                p_PLLOUT_SELECT = "GENCLK",
+                p_ENABLE_ICEGATE = 0,
+                # IO
+                i_REFERENCECLK = clk16,
+                o_PLLOUTCORE = self.cd_usb_48.clk,
+                #o_PLLOUTGLOBAL,
+                #i_EXTFEEDBACK,
+                #i_DYNAMICDELAY,
+                #o_LOCK,
+                i_BYPASS = 0,
+                i_RESETB = 1,
+                #i_LATCHINPUTVALUE,
+                #o_SDO,
+                #i_SDI,
+            ),
+        ]
+
+
+class RAMSoC(SoCCore):
+    def __init__(self, platform, **kwargs):
+
+        # The TinyFPGA BX has 128kbit of blockram == 16kbytes
+        # Each BRAM is 512bytes == 32 BRAM blocks
+        # Need 4 BRAM blocks for USB, leaving 28.
+        #  - 8kbytes ROM - 16 x BRAMs
+        #  - 4kbytes RAM -  8 x BRAMs
+
+        kwargs['integrated_rom_size']=0
+        kwargs['integrated_sram_size']=16*1024
+
+        # Disable BIOS
+        #kwargs['integrated_rom_init'] = [0]
+
+        # FIXME: Force either lite or minimal variants of CPUs; full is too big.
+        if 'cpu_variant' not in kwargs:
+            kwargs['cpu_variant']='lite'
+
+        platform.add_extension(serial)
+        platform.add_extension(reset_button)
+        clk_freq = int(16e6)
+
+        # Extra 0x28000 is due to bootloader bitstream.
+        SoCCore.__init__(self, platform, clk_freq, **kwargs)
+
+        self.submodules.crg = _CRG(platform)
+        self.platform.add_period_constraint(self.crg.cd_sys.clk, 1e9/clk_freq)
+
+        self.submodules.info = info.Info(platform, self.__class__.__name__)
+        self.add_csr("info")
+
+        # Control and Status
+#        self.submodules.cas = cas.ControlAndStatus(platform, clk_freq)
+#        self.add_csr("cas")
+
+        # SPI flash peripheral
+        self.submodules.spiflash = spi_flash.SpiFlashSingle(
+            platform.request("spiflash"),
+            dummy=platform.spiflash_read_dummy_bits,
+            div=platform.spiflash_clock_div)
+        self.add_constant("SPIFLASH_PAGE_SIZE", platform.spiflash_page_size)
+        self.add_constant("SPIFLASH_SECTOR_SIZE", platform.spiflash_sector_size)
+        self.register_mem("spiflash", self.mem_map["spiflash"],
+            self.spiflash.bus, size=platform.spiflash_total_size)
+
+        bios_size = 0x8000
+        self.add_constant("ROM_DISABLE", 1)
+        self.add_memory_region("rom", kwargs['cpu_reset_address'], bios_size)
+        self.flash_boot_address = self.mem_map["spiflash"]+platform.gateware_size+bios_size+platform.bootloader_size
+
+        # Disable USB activity until we switch to a USB UART.
+        #self.comb += [platform.request("usb").pullup.eq(0)]
+
+        # Arachne-pnr is unsupported- it has trouble routing this design
+        # on this particular board reliably. That said, annotate the build
+        # template anyway just in case.
+        # Disable final deep-sleep power down so firmware words are loaded
+        # onto softcore's address bus.
```

Comment:
This should call `platform_toolchain_extend` I think?

Reply:
`-s` is now passed to `icepack` in LiteX by default, so this update is not needed anymore